### PR TITLE
feat(http): add support for blob as a response type

### PR DESCRIPTION
**Pull request by cassand**
**Please check if the PR fulfills these requirements**
- [ x ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

```
[ ] Bugfix
[ x ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

**What is the current behavior?** 
There is no way to receive a response as a blob.

**What is the new behavior?**
Allow to set the attribute responseType of the request, through the enum ResponseContentType.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

**Other information**:

In order to add appropriate tests for this feature I have to introduce some change into the MockBrowserXHR, (which will also allow to test arraybuffer, json and text as a responseType), therefore I will add them in a future commit.

### CODE REVIEW

**Comment by zoechi** [comment]
dup of angular/http#75 ?

File: tools/public_api_guard/http/index.d.ts [hunk 1]
```diff
@@ -179,14 +179,15 @@ export declare class Response extends Body {
 
 /** @experimental */
 export declare enum ResponseContentType {
-    ArrayBuffer = 0,
```
Comment (vicb) [inline tools/public_api_guard/http/index.d.ts]: Why do we need to explicitly assign value here ?

**Comment by vicb** [comment]
Please add tests. Thanks

File: tools/public_api_guard/http/index.d.ts (hunk 1 above)
Comment (cassand) [inline tools/public_api_guard/http/index.d.ts]: This is generated by the command `gulp public-api:update`, if the values are not explicitly assigned, then any call to `gulp public-api:update` will put back the values.

Comment (vicb) [inline tools/public_api_guard/http/index.d.ts]: you're right. didn't notice it was a .d.ts file before

**Comment by vicb** [comment]
@cassand please merge both the PRs and add tests. Thanks

**Comment by cassand** [comment]
Both requests are merged and the feature is tested.

**Comment by tomalex0** [comment]
I'm forced to ask here as i don't got any response  from  gitter. 

Any example showcasing sending  `FormData` as part of `http`,  

Below code  works fine for me but like to know if same supported in `http`
```
 var files = $event.target.files || $event.srcElement.files;
        var file = files[0];
        let formData = new FormData();
        formData.append("single_fileup", file);
        formData.append('key1', 'value1');
        formData.append('key2', 'value2');
       var req = new XMLHttpRequest();
       req.open("POST", "/api/fileupload");
       req.send(formData);
```

```
var multer  = require('multer');
var storage = multer.memoryStorage();
var upload = multer({ storage: storage });
  router.post('/api/fileupload', upload.single('single_fileup'), function(req, res, next){

});
```

How to make below code work?
```
 this.http.post('/api/fileupload', formData)
            .map(this.extractData)
            .catch(this.handleError);
```

**Pull request merged**